tests/test_mailchimp_discovery.py

```diff
@@ -0,0 +1,159 @@
+        found_catalog_names = {c["tap_stream_id"] for c in found_catalogs}
+        self.assertTrue(
+            all([re.fullmatch(r"[a-z_]+", name) for name in found_catalog_names]),
+            msg="One or more streams don't follow standard naming",
+        )
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+
+                # Verify the catalog is found for a given stream
+                catalog = next(
+                    iter(
+                        [
+                            catalog
+                            for catalog in found_catalogs
+                            if catalog["stream_name"] == stream
+                        ]
+                    )
+                )
+                self.assertIsNotNone(catalog)
+
+                # Collecting expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_automatic_fields = self.expected_automatic_fields().get(stream)
+                expected_replication_method = self.expected_replication_method()[stream]
+
+                # Collecting actual values...
+                schema_and_metadata = menagerie.get_annotated_schema(
+                    conn_id, catalog["stream_id"]
+                )
+                metadata = schema_and_metadata["metadata"]
+
+                # BUG: TDL-20301 EMPTY BREADCRUMB IS NOT GENERATED
+                # stream_properties = [
+                #     item for item in metadata if item.get("breadcrumb") == []]
+
+                # actual_primary_keys = set(
+                #     stream_properties[0].get(
+                #         "metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, [])
+                # )
+                # actual_replication_keys = set(
+                #     stream_properties[0].get(
+                #         "metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, [])
+                # )
+                # actual_replication_method = stream_properties[0].get(
+                #     "metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
+
+                actual_automatic_fields = set(
+                    item.get("breadcrumb", ["properties", None])[1]
+                    for item in metadata
+                    if item.get("metadata").get("inclusion") == "automatic"
+                )
+
+                actual_fields = []
+                for md_entry in metadata:
+                    if md_entry["breadcrumb"] != []:
+                        actual_fields.append(md_entry["breadcrumb"][1])
+
+                ##########################################################################
+                # metadata assertions
+                ##########################################################################
+
+                # BUG: TDL-20301 EMPTY BREADCRUMB IS NOT GENERATED
```

Comment:
TDL-20301 is resolved, uncomment the code.

Reply:
done